include/openssl/params.h

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2019 The OpenSSL Project Authors. All Rights Reserved.
+ * Copyright (c) 2019, Oracle and/or its affiliates.  All rights reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#ifndef HEADER_PARAMS_H
+# define HEADER_PARAMS_H
+
+#include <openssl/core.h>
+#include <openssl/bn.h>
+
+int ossl_param_get_int8(const OSSL_PARAM *p, const char *key, int8_t *val);
+int ossl_param_get_uint8(const OSSL_PARAM *p, const char *key, uint8_t *val);
+int ossl_param_get_int16(const OSSL_PARAM *p, const char *key, int16_t *val);
+int ossl_param_get_uint16(const OSSL_PARAM *p, const char *key, uint16_t *val);
+int ossl_param_get_int32(const OSSL_PARAM *p, const char *key, int32_t *val);
+int ossl_param_get_uint32(const OSSL_PARAM *p, const char *key, uint32_t *val);
+int ossl_param_get_int64(const OSSL_PARAM *p, const char *key, int64_t *val);
+int ossl_param_get_uint64(const OSSL_PARAM *p, const char *key, uint64_t *val);
+int ossl_param_get_int(const OSSL_PARAM *p, const char *key, int *val);
+int ossl_param_get_uint(const OSSL_PARAM *p, const char *key,
+                        unsigned int *val);
+int ossl_param_get_char(const OSSL_PARAM *p, const char *key, char *val);
+int ossl_param_get_uchar(const OSSL_PARAM *p, const char *key,
+                         unsigned char *val);
+int ossl_param_get_short(const OSSL_PARAM *p, const char *key, short int *val);
+int ossl_param_get_ushort(const OSSL_PARAM *p, const char *key,
+                          unsigned short int *val);
+int ossl_param_get_long(const OSSL_PARAM *p, const char *key,
+                        long int *val);
+int ossl_param_get_ulong(const OSSL_PARAM *p, const char *key,
+                         unsigned long int *val);
+int ossl_param_get_long_long(const OSSL_PARAM *p, const char *key,
```

Comment:
Are the long long types needed? (Long long means mental in pidgin)

Reply:
Anyone have thoughts on this?

Reply:
I've removed the long long calls.  It is down to sign/unsigned int & long and the specified length integers.